src/rax.c

```diff
@@ -158,32 +160,55 @@ static inline void raxStackFree(raxStack *ts) {
  * If datafield is true, the allocation is made large enough to hold the
  * associated data pointer.
  * Returns the new node pointer. On out of memory NULL is returned. */
-raxNode *raxNewNode(size_t children, int datafield) {
+raxNode *raxNewNode(rax *rax, size_t children, int datafield) {
     size_t nodesize = sizeof(raxNode)+children+raxPadding(children)+
                       sizeof(raxNode*)*children;
     if (datafield) nodesize += sizeof(void*);
-    raxNode *node = rax_malloc(nodesize);
+    size_t usable;
+    raxNode *node = rax_malloc_usable(nodesize,&usable);
     if (node == NULL) return NULL;
     node->iskey = 0;
     node->isnull = 0;
     node->iscompr = 0;
     node->size = children;
+    if (rax->flags & RAX_ACCOUNT_ALLOC_SIZE) {
+        size_t *alloc_size = (size_t *)rax->metadata;
+        *alloc_size += usable;
+    }
     return node;
 }
 
+/* Deallocate node */
+void raxFreeNode(rax *rax, raxNode *n) {
+    size_t usable;
+    rax_free_usable(n, &usable);
+    if (rax->flags & RAX_ACCOUNT_ALLOC_SIZE) {
+        size_t *alloc_size = (size_t *)rax->metadata;
+        *alloc_size -= usable;
+    }
+}
+
 /* Allocate a new rax and return its pointer. On out of memory the function
  * returns NULL. */
 rax *raxNew(void) {
-    return raxNewWithMetadata(0);
+    return raxNewWithMetadata(0, 0);
 }
 
 /* Allocate a new rax with metadata */
-rax *raxNewWithMetadata(int metaSize) {
-    rax *rax = rax_malloc(sizeof(*rax) + metaSize);
+rax *raxNewWithMetadata(int metaSize, int flags) {
+    assert(!(flags & RAX_ACCOUNT_ALLOC_SIZE) ||
+           (size_t)metaSize >= sizeof(size_t));
```

Comment:
is metaSize meaningless to be larger than sizeof(size_t)?
```suggestion
           (size_t)metaSize == sizeof(size_t));
```

Reply:
Potentially it's useful. Currently hashes do this where they start with `sizeof(size_t)` metadata and if expirable keys are present are switched to have expire metadata (with first member `size_t alloc_size`, preserving the previous value).